### ISSUE TITLE
Images are not decrypted

### ISSUE THREAD

**Issue by Natureshadow** [stv0g/transwhat#86]
In teh feature list, it says that images are supported. However, I aonly get a link to the encrypted image file on WhatsApp's HttpUpload component.

How to handle it?

**Comment by stv0g** [comment]
I think the it worked back then when images weren’t encrypted yet…

Nowadays handling encrypted files is more difficult as transWhat would need to decrypt them itself and then pass them in plaintext to Spectrum.

For now, we don’t support filetransfers.

**Comment by Natureshadow** [comment]
Well, I don't no if Spectrum gets in the way there, but as long as the XMPP server has HttpUpload, decrypting the file with axolotl and then jsut pushing it there seems quite easy.

**Comment by Natureshadow** [comment]
Also, it looks like yowsup supports it:
https://github.com/tgalal/yowsup/issues/2235

So, the problem looks solvable. Would you welcome a pull request if I tried implementing it?

**Comment by stv0g** [comment]
Sure, pull requests are always welcome! It would be cool to support HttpUpload!

We already had a discussion to support HttpUpload.
But then we went for another easier solution: https://github.com/stv0g/transwhat/issues/54

**Comment by Natureshadow** [comment]
Yeah, that's fine as well. Didn't know Spectrum had a web directory thing as well.

So, the issue seems to be that WhatsApp changed the way media is encrypted: https://github.com/tgalal/yowsup/issues/1400

The state of the fix in yowsup is not very clear…

**Comment by dazzzl** [comment]
I have used the jlguardi's yowsup branch for this. In fact it is still working. However, my yowsup is now a (partly outdated) mixture of  jlguardi's and the official branch. But, it is working for me ....
